pkg/data/redis.go

```diff
@@ -0,0 +1,458 @@
+		return opts.insertDataInRedisCluster(rows)
+	}
+
+	redisCommand := []interface{}{
+		"eval", dataInsertScript, "0", fmt.Sprintf("%d", rows), "hash",
+	}
+	output, err := opts.execCommand("", redisCommand)
+	if err != nil {
+		return err
+	}
+	if output != successfulScriptReturn {
+		fmt.Printf("Error. Can not insert data in master node. Output: %s\n", output)
+	}
+	fmt.Printf("\n%d keys inserted in redis database %s/%s successfully\n", rows, opts.db.Namespace, opts.db.Name)
+	return nil
+}
+
+func (opts *redisOpts) insertDataInRedisCluster(rows int) error {
+	var slotKey map[int64]string
+	err := json.Unmarshal(redisutil.ClusterSlotKeys, &slotKey)
+	if err != nil {
+		return err
+	}
+	masterNodes, err := opts.getClusterMasterNodes()
+	if err != nil {
+		return err
+	}
+	keysPerMaster := rows / len(masterNodes)
+	extraKey := rows % len(masterNodes)
+	for _, node := range masterNodes {
+		keyHash := slotKey[node.slot]
+		redisCommand := []interface{}{
+			"eval", dataInsertScript, "0", fmt.Sprintf("%d", keysPerMaster+extraKey), keyHash,
+		}
+		extraKey = 0
+		output, err := opts.execCommand(node.host, redisCommand)
+		if err != nil {
+			return err
+		}
+		if output != successfulScriptReturn {
+			fmt.Printf("Error. Can not insert data in master %s. Output: %s\n", node.host, output)
+		}
+	}
+	fmt.Printf("\n%d keys inserted in redis database %s/%s successfully\n", rows, opts.db.Namespace, opts.db.Name)
+	return nil
+}
+
+func VerifyRedisDataCMD(f cmdutil.Factory) *cobra.Command {
+	var (
+		dbName string
+		rows   int
+	)
+
+	rdVerifyCmd := &cobra.Command{
+		Use: "redis",
+		Aliases: []string{
+			"rd",
+		},
+		Short:   "Verify rows in a redis database",
+		Long:    `Use this cmd to verify data in a redis object`,
+		Example: `kubectl dba verify redis -n demo sample-redis --rows 500`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				log.Fatal("Enter redis object's name as an argument. Your commands will be applied on a database inside it's primary pod")
+			}
+			dbName = args[0]
+
+			namespace, _, err := f.ToRawKubeConfigLoader().Namespace()
+			if err != nil {
+				klog.Error(err, "failed to get current namespace")
+			}
+
+			opts, err := newRedisOpts(f, dbName, namespace)
+			if err != nil {
+				log.Fatalln(err)
+			}
+
+			err = opts.verifyRedisData(rows)
+			if err != nil {
+				log.Fatalln(err)
+			}
+		},
+	}
+
+	rdVerifyCmd.Flags().IntVarP(&rows, "rows", "r", 10, "rows in ")
+
+	return rdVerifyCmd
+}
+
+func (opts *redisOpts) verifyRedisData(rows int) error {
+	if opts.db.Spec.Mode == api.RedisModeCluster {
+		return opts.verifyDataInRedisCluster(rows)
+	}
+	redisCommand := []interface{}{
+		"dbsize",
+	}
+	output, err := opts.execCommand("", redisCommand)
+	if err != nil {
+		return err
+	}
+	totalKeys, err := strconv.Atoi(output)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("\nExpected keys: %d .Redis database %s/%s contains: %d keys\n", rows, opts.db.Namespace, opts.db.Name, totalKeys)
```

Comment:
if rows doesn't matched totalKey, we can show that. But if they matched, we should show some `Success` type message to the user.

Reply:
Should it be exact match ? Because in Redis, we insert some keys for health check. That also comes when we take the total count. What about greater or equal ?